requirements.in

```diff
@@ -2,11 +2,15 @@
 #
 # Generates requirements.txt file.
 
-boto3==1.28.68
-botocore==1.31.68
 dockerflow==2022.8.0
-google-auth==2.25.2
-google-api-core[grpc]==2.15.0
-google-cloud-logging==3.9.0
-googleapis-common-protos[grpc]==1.62.0
+google-auth==2.29.0
+google-api-core[grpc]==2.19.0
+google-cloud-logging==3.10.0
+googleapis-common-protos[grpc]==1.63.0
 requests==2.31.0
+
+# NOTE(willkg): These need to be locked together. This is a known-good
+# combination.
+boto3==1.28.68
+botocore==1.31.68
+s3transfer==0.7.0
```

Comment:
They made some changes in botocore and s3transfer and we need to keep these three dependencies in lockstep, so I added them all to the `requirements.in` file with a note so future me doesn't spend a bunch of time being puzzled by this situation:

https://github.com/boto/botocore/issues/3082